cherab/solps/solps_plasma.py

```diff
@@ -17,228 +17,577 @@
+            velocities[k, :2] = self.mesh.to_poloidal(value[k, [0, 2]])
+
+        self._velocities_cylindrical = value
+        self._velocities = velocities
+        self._velocities_cylindrical_f2d = {}
+        self._velocities_cartesian = {}
+        for k, sp in enumerate(self._species_list):
+            self._velocities_cylindrical_f2d[k] = SOLPSVectorFunction2D.instance(self._sample_vector_f2d, value[k])
+            self._velocities_cylindrical_f2d[sp] = self._velocities_cylindrical_f2d[k]
+            self._velocities_cartesian[k] = VectorAxisymmetricMapper(self._velocities_cylindrical_f2d[k])
+            self._velocities_cartesian[sp] = self._velocities_cartesian[k]
+
+    @property
+    def velocities_cylindrical_f2d(self):
+        """
+        Dictionary of VectorFunction2D interpolators for species velocities
+        in cylindrical coordinates.
+        Accessed by species index or species_list elements.
+        E.g., elocities_cylindrical_f2d[1] or elocities_cylindrical_f2d[('deuterium', 1))].
+        Each entry returns a vector of species velocity at a given point (R, Z).
+        :return:
+        """
+        return self._velocities_cylindrical_f2d
 
     @property
     def velocities_cartesian(self):
+        """
+        Dictionary of VectorFunction3D interpolators for species velocities
+        in Cartesian coordinates.
+        Accessed by species index or species_list elements.
+        E.g., elocities_cylindrical_f3d[1] or elocities_cylindrical_f3d[('deuterium', 1))].
+        Each entry returns a vector of species velocity at a given point (x, y, z).
+        :return:
+        """
         return self._velocities_cartesian
 
+    @property
+    def inside_mesh(self):
+        """
+        Function2D for testing if point p is inside the simulation mesh.
+        """
+        return self._inside_mesh
+
     @property
     def inside_volume_mesh(self):
         """
         Function3D for testing if point p is inside the simulation mesh.
         """
-        if self._inside_mesh is None:
-            raise RuntimeError("Inside mesh test not available for this simulation")
-        else:
-            return self._inside_mesh
+        return AxisymmetricMapper(self._inside_mesh)
 
     @property
     def total_radiation(self):
         """
-        Total radiation
+        Total radiation at each mesh cell.
+        Array of shape (ny, nx).
 
         This is not calculated from the CHERAB emission models, instead it comes from the SOLPS output data.
         Is calculated from the sum of all integrated line emission and all Bremmstrahlung. The signals used are 'RQRAD'
         and 'RQBRM'. Final output is in W/str?
         """
-        if self._total_rad is None:
-            raise RuntimeError("Total radiation not available for this simulation")
+        if self._total_radiation is None:
+            raise RuntimeError("Total radiation not available for this simulation.")
         else:
```

Comment:
`else` is not necessary after a `raise`. This applies to all the `total_radiation*`, `b_field*` and `eirene_simulation` properties.

Reply:
This code was before me, some properties threw errors when not initialized and some did not. I think, they should not throw errors. If a property is required for a function to work, that function must check if the value is set and throw a `RuntimeError` if not. I've removed these `raise` statements from properties and instead added them to functions that don't work without these properties.